scripts/build/clean.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Cleans top-level build directories (for OSS and internal processes),
+ * dist subfolders under packages, and .js[.map] files in package directories.
+ */
+
+const {exec} = require('child_process');
+const fs = require('fs');
+
+const {sync: globSync} = require('glob');
+
+function main() {
+  removeDirectory('build');
+  removeDirectory('.rewrite-tmp');
+  removeFilesOfType('css');
+  removeFilesOfType('js');
+  removeFilesOfType('d.ts');
+  removeFilesOfType('map');
+}
+
+function removeDirectory(directory) {
+  exec(`rm -rf ${directory}`, (err) => {
+    if (err) {
+      throw err;
+    }
+  });
+}
+
+function removeFilesOfType(type) {
+  const fileGlob = `packages/*/*.${type}`;
+  const nestedFileGlob = `packages/*/!(node_modules)/**/*.${type}`;
```

Comment:
Remove the negation and pass `{ignore: ['**/node_modules/**']}` to `globSync()` as the second param.

Reply:
Oh wow - that's so much better. I was thinking I made this too complicated.